Sdk@3.0.0 preview.0

vmarta opened the pull request:
Todos:
- [x] Set target branch to `release-3.0.x`
- [ ] Make a similar PR for `commerce-sdk-react@1.0.0-preview.0`
- [ ] Make a similar PR for `retail-react-app@1.0.0-preview.0`

Related PR #1233 

# Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] **Bug fix** (non-breaking change that fixes an issue)
- [ ] **New feature** (non-breaking change that adds functionality)
- [ ] **Documentation update**
- [ ] **Breaking change** (could cause existing functionality to not work as expected)
- [ ] **Other changes** (non-breaking changes that does not fit any of the above)

> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- (step1)

# Checklists

<!--- Enter an `x` in all the boxes that apply. -->
<!--- If you’re unsure about any of these, don’t hesitate to ask. We’re here to help! -->

## General

- [ ] Changes are covered by test cases
- [ ] CHANGELOG.md updated with a short description of changes (_not_ required for documentation updates)

## Accessibility Compliance

You must check off all items in **one** of the follow two lists:

- [ ] There are no changes to UI

_or..._

- [ ] Changes were tested with a Screen Reader (iOS VoiceOver or Android Talkback) and had no issues
- [ ] Changes comply with [WCAG 2.0 guidelines levels A and AA](https://www.wuhcag.com/wcag-checklist/)
- [ ] Changes to common UI patterns and interactions comply with [WAI-ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## Localization

- [ ] Changes include a UI text update in the Retail React App (which requires translation)

bfeister committed:
fix package lockfiles that were out of sync

bfeister committed:
Revert "fix package lockfiles that were out of sync"

This reverts commit 7f3c3645ef963ba4d7a7c219e66ec93ac6fe2269.

yunakim714 committed:
progress

bfeister committed:
fix out of sync lockfiles

bfeister committed:
Merge branch 'v3' into feature/template-extensibility

* v3:
  Drop npm 6 and node 14 from V3 (#1112)
  Upgrade lerna (#1111)
  add locale and currency to apiClients (#1107)
  Update test.yml (#1109)
  [V3] Invoke push action for v3 to `staging-v3` target (#1108)
  remove commerce api folder (#1105)

# Conflicts:
#	package-lock.json
#	package.json
#	packages/commerce-sdk-react/package-lock.json
#	packages/commerce-sdk-react/package.json
#	packages/internal-lib-build/package-lock.json
#	packages/pwa-kit-create-app/package-lock.json
#	packages/pwa-kit-create-app/package.json
#	packages/pwa-kit-dev/package-lock.json
#	packages/pwa-kit-dev/package.json
#	packages/pwa-kit-react-sdk/package-lock.json
#	packages/pwa-kit-runtime/package-lock.json
#	packages/template-retail-react-app/app/mocks/mock-data.js
#	packages/template-retail-react-app/package-lock.json
#	packages/template-typescript-minimal/package-lock.json
#	packages/test-commerce-sdk-react/package-lock.json

yunakim714 committed:
merge feature branch back

yunakim714 committed:
resolve merge conflicts

bfeister committed:
Merge branch 'v3' into feature/template-extensibility

* v3:
  Sort contents of package.json files. (#1115)
  Fail fast? False! (#1116)

# Conflicts:
#	packages/pwa-kit-create-app/package.json
#	packages/pwa-kit-dev/package.json
#	packages/pwa-kit-runtime/package.json
#	packages/template-typescript-minimal/package.json

bfeister committed:
merge conflict

bfeister committed:
resolve merge conflict

bfeister committed:
update spike project's engines / regen v3 lockfiles

yunakim714 committed:
regenerate lock files

bfeister committed:
add missing node engines

bfeister committed:
add missing useToast (bad merge)

bfeister committed:
updates from underlying template-retail-react-app

bfeister committed:
drop unnecessary pages/product-detail override

bfeister committed:
Merge branch 'feature/template-extensibility' into extensibility-code-rehaul

* feature/template-extensibility:
  drop unnecessary pages/product-detail override
  updates from underlying template-retail-react-app
  add missing useToast (bad merge)
  add missing node engines
  update spike project's engines / regen v3 lockfiles
  resolve merge conflict
  merge conflict

# Conflicts:
#	packages/spike-extendend-retail-app/pwa-kit/overrides/app/components/header/index.jsx
#	packages/spike-extendend-retail-app/pwa-kit/overrides/app/pages/product-detail/index.jsx

bfeister committed:
fix dupe useToast

bfeister committed:
progress on making `charles-solution` work

bfeister committed:
initial pass at migration to `resolver.getHook('resolve').tapAsync` working, TODO next... cleanup

yunakim714 committed:
restore icons file and add extensible text to home

yunakim714 committed:
add some comments and construct hashmap in new plugin

yunakim714 committed:
working glob sync lookup??

yunakim714 committed:
resolve lookup when there is no extension

yunakim714 committed:
add component slot for testing

yunakim714 committed:
fix some lookup errors

bfeister committed:
migrate to new directory

bfeister committed:
remove unnecessary directory nesting, remove unnecessary files for "minimum file project"

bfeister committed:
add example of extending routes non-destructively, add example of quickly toggling some base constant values (e.g. categories shown on home page)

bfeister committed:
Update example route

bfeister committed:
add some demo examples of overrides

yunakim714 committed:
cleanup

bfeister committed:
commit package name change

yunakim714 committed:
rewrite requests coming from underlying template

yunakim714 committed:
cleanup

yunakim714 committed:
rename plugin file

yunakim714 committed:
resolve merge conflict

yunakim714 committed:
remove unnecessary build files

yunakim714 committed:
rename to overrides

yunakim714 committed:
fix lookup errors

bfeister committed:
Merge branch 'v3' into feature/template-extensibility

* v3:
  [Spike]Replace watch with nodemon (#1146)
  Fix Page Designer ImageWithText Link component (#1092) (#1148)
  Upgrade deprecated dependencies (#1124)
  Restart login flow if refresh_token is invalid (#1135)
  Move some util to site-utils to avoid circular imports (#1133)
  Restore old file name. (#1140)
  Update eslint configuration (#1129)
  CI: clarify the environment variables (#1127)
  Remove react-query-devtools from production build (#1121)
  Update lerna.json (#1118)
  Parallelize lighthouse ci (#1126)
  Increase test timeouts only on CI env (#1123)
  Remove unused `request` deprecated dependency. (#1125)
  Upgrade msw to latest (#1100)
  Add mergeBasket hook (#1114)
  [V3][Hooks Integration 🪝] Manually update cache for ShopperCustomer (#1113)
  dont use callback on mutateAsync (#1119)
  2-spaces not 4-spaces (#1117)

# Conflicts:
#	packages/internal-lib-build/package-lock.json
#	packages/pwa-kit-create-app/package-lock.json
#	packages/pwa-kit-dev/package-lock.json
#	packages/pwa-kit-dev/package.json
#	packages/pwa-kit-dev/src/configs/webpack/config.js
#	packages/pwa-kit-dev/src/ssr/server/build-dev-server.js
#	packages/pwa-kit-react-sdk/package-lock.json
#	packages/pwa-kit-runtime/package-lock.json
#	packages/template-express-minimal/package-lock.json
#	packages/template-mrt-reference-app/package-lock.json
#	packages/template-retail-react-app/package-lock.json
#	packages/template-typescript-minimal/package-lock.json
#	packages/test-commerce-sdk-react/package-lock.json

bfeister committed:
Merge branch 'feature/template-extensibility' into extensibility-code-rehaul

* feature/template-extensibility:
  [Spike]Replace watch with nodemon (#1146)
  Fix Page Designer ImageWithText Link component (#1092) (#1148)
  Upgrade deprecated dependencies (#1124)
  Restart login flow if refresh_token is invalid (#1135)
  Move some util to site-utils to avoid circular imports (#1133)
  Restore old file name. (#1140)
  Update eslint configuration (#1129)
  CI: clarify the environment variables (#1127)
  Remove react-query-devtools from production build (#1121)
  Update lerna.json (#1118)
  Parallelize lighthouse ci (#1126)
  Increase test timeouts only on CI env (#1123)
  Remove unused `request` deprecated dependency. (#1125)
  Upgrade msw to latest (#1100)
  Add mergeBasket hook (#1114)
  [V3][Hooks Integration 🪝] Manually update cache for ShopperCustomer (#1113)
  dont use callback on mutateAsync (#1119)
  2-spaces not 4-spaces (#1117)

# Conflicts:
#	packages/internal-lib-build/package-lock.json
#	packages/pwa-kit-dev/package-lock.json
#	packages/pwa-kit-react-sdk/package-lock.json
#	packages/pwa-kit-runtime/package-lock.json
#	packages/pwa-kit-runtime/package.json

bfeister committed:
fix some broken upstream changes, eslint fixes

bfeister committed:
fix bad merge conflict

vmarta committed:
Bring back the deleted package.json

vmarta committed:
bring back deleted package lockfiles

vmarta committed:
Install new eslint plugin

vmarta committed:
No more relative imports in retail-react-app

vmarta committed:
Create .prettierrc.yaml

bfeister committed:
restore other deleted package lockfiles

bfeister committed:
fix some eslint issues

bfeister committed:
remove `^` imports and begin migrating logic away from previous API. TODO: get webpack to resolve `retail-react-app` base import path from within `template-retail-react-app`

bfeister committed:
short circuit plugin temporarily, get webpack `alias` working for template to resolve self-referencing `retail-react-app` imports

bfeister committed:
fix unreachable condition

bfeister committed:
wow... freaking works 🚀

bfeister committed:
cleanup, get `template-retail-react-app` working again

bfeister committed:
cleanup

vmarta committed:
Create .eslintrc.js

bfeister committed:
fix routes.jsx which no longer pulled relative template

bfeister committed:
make routes.jsx more concise

bfeister committed:
update lockfiles

bfeister committed:
add jsconfig.json for IDEs to have magic `retail-react-app` be discoverable

bfeister committed:
Merge branch 'v3' into feature/template-extensibility

* v3:
  split ssr build on local (#1155) (#1156)
  [V3] Update Page Designer (#1128)
  [V3][Hooks Integration 🪝] Replace last of the `getProps` (#1149)
  Fix Page Designer ImageWithText Link component (#1092)

# Conflicts:
#	packages/template-retail-react-app/app/components/_app-config/index.jsx

bfeister committed:
Merge branch 'feature/template-extensibility' into feature/template-extensibility-algo-refactor

* feature/template-extensibility:
  split ssr build on local (#1155) (#1156)
  [V3] Update Page Designer (#1128)
  bring back deleted package lockfiles
  fix bad merge conflict
  [V3][Hooks Integration 🪝] Replace last of the `getProps` (#1149)
  Fix Page Designer ImageWithText Link component (#1092)

# Conflicts:
#	packages/internal-lib-build/package-lock.json
#	packages/pwa-kit-create-app/package-lock.json
#	packages/template-express-minimal/package-lock.json
#	packages/template-mrt-reference-app/package-lock.json
#	packages/template-retail-react-app/app/components/_app-config/index.jsx
#	packages/template-retail-react-app/app/components/_app/index.jsx
#	packages/template-retail-react-app/app/page-designer/core/component/index.jsx
#	packages/template-retail-react-app/app/page-designer/core/component/index.test.js
#	packages/template-retail-react-app/app/page-designer/core/page/index.jsx
#	packages/template-retail-react-app/app/page-designer/core/page/index.test.js
#	packages/template-retail-react-app/app/page-designer/core/region/index.jsx
#	packages/template-retail-react-app/app/page-designer/core/region/index.test.js
#	packages/template-retail-react-app/app/page-designer/layouts/carousel/index.jsx
#	packages/template-retail-react-app/app/page-designer/layouts/mobileGrid1r1c/index.jsx
#	packages/template-retail-react-app/app/page-designer/layouts/mobileGrid2r1c/index.jsx
#	packages/template-retail-react-app/app/page-designer/layouts/mobileGrid2r2c/index.jsx
#	packages/template-retail-react-app/app/page-designer/layouts/mobileGrid2r3c/index.jsx
#	packages/template-retail-react-app/app/page-designer/layouts/mobileGrid3r1c/index.jsx
#	packages/template-retail-react-app/app/page-designer/layouts/mobileGrid3r2c/index.jsx
#	packages/template-retail-react-app/app/pages/page-not-found/index.jsx
#	packages/template-retail-react-app/app/utils/test-utils.js
#	packages/template-typescript-minimal/package-lock.json
#	packages/test-commerce-sdk-react/package-lock.json

bfeister committed:
more cleanup

bfeister committed:
change package.json key from `mobify` => `ccExtensibility`

bfeister committed:
cleanup

bfeister committed:
Final cleanup, remove console.logs, make `extendable` dynamic for internal file resolution in `template-retail-react-app` (and future projects), remove some non-required files,

bfeister committed:
fix straggler non-dynamic filepath

bfeister committed:
cleanup

bfeister committed:
remove code that shouldn't be in v3+

vmarta committed:
bump-version script ignores the retail-react-app

vmarta committed:
See if this `--pre-dist-tag` works as expected

vmarta committed:
Clean up code

vmarta committed:
Add todo

bfeister committed:
sync package.json deps in extended example project

bfeister committed:
cleanup / pr feedback

vmarta committed:
Update npm scripts

vmarta committed:
Disable `npm install` step

vmarta committed:
List the packages' versions for easy review

vmarta committed:
Fix bug with versioning the root package file

vmarta committed:
Merge branch 'feature/template-extensibility-algo-refactor' into versioning-retail-react-app

vmarta committed:
A bit of refactoring

vmarta committed:
Some clean up

bfeister committed:
pr feedback, great catch @kevinxh

bfeister committed:
fix problem with bad copy from underlying template file

bfeister committed:
pr feedback / cleanup

bfeister committed:
cleanup

Co-authored-by: Kevin He <kevin.he@salesforce.com>

vmarta committed:
Merge branch 'feature/template-extensibility-algo-refactor' into versioning-retail-react-app

vmarta committed:
Update package-lock.json

vcua-mobify committed:
WIP

bfeister committed:
drop unneeded complexity

bfeister committed:
lockfiles

bfeister committed:
moving toward an array of `alias`es

vmarta committed:
Generator: try downloading retail-react-app@latest from npm

vmarta committed:
Some refactoring

bfeister committed:
Refactor template extensibility functionality to use `resolve.alias` + greatly reduce code footprint + make leading slash optional in `ccExtensibility.overridesDir`

vmarta committed:
Tweak the versioning logic

vmarta committed:
These are optional options

vmarta committed:
Use a new npm tag for testing purpose

vcua-mobify committed:
Merge branch 'feature/template-extensibility-algo-refactor' of github.com:SalesforceCommerceCloud/pwa-kit into feature/template-extensibility-algo-refactor

bfeister committed:
add brand-logo with no clash for server side viewBox

bfeister committed:
add `request-processor.js` (required for worker.js to load) and fix bad extensibility path loading in `build-dev-server.js`

vmarta committed:
Merge branch 'feature/template-extensibility-algo-refactor' into versioning-retail-react-app

vmarta committed:
Add some comments

vmarta committed:
Install shelljs

vmarta committed:
retail-react-app: version updates scoped to this directory only

vmarta committed:
Update versioning logic in the root

vmarta committed:
retail-react-app: simplify the versioning logic

vmarta committed:
fake sdk release v3

vmarta committed:
fake release of retail-react-app v1

bfeister committed:
lint fix

bfeister committed:
fix failing pwa-kit-react-sdk tests

bfeister committed:
cleanup

bfeister committed:
Merge branch 'v3' into feature/template-extensibility

* v3:
  fix: remove device-context and detect-device-type (#1168)
  Dependency updates (#1170)
  [Snyk] Security upgrade cosmiconfig from 7.1.0 to 8.0.0 (#1145)
  [V3] Remove `cross-fetch` from `template-retail-react-app` (#1160)
  [v3] Add suffix to ssr build files (#1158)
  Consume new version of the commerce-sdk-isomorphic to enable Phased Launches support HTTP Basic Auth (#1153)

# Conflicts:
#	packages/commerce-sdk-react/package-lock.json
#	packages/internal-lib-build/package-lock.json
#	packages/pwa-kit-runtime/package-lock.json
#	packages/template-retail-react-app/package-lock.json

bfeister committed:
regen package lockfiles

bfeister committed:
Merge branch 'feature/template-extensibility' into feature/template-extensibility-algo-refactor + regen lockfiles

* feature/template-extensibility:
  regen package lockfiles
  fix: remove device-context and detect-device-type (#1168)
  Dependency updates (#1170)
  [Snyk] Security upgrade cosmiconfig from 7.1.0 to 8.0.0 (#1145)
  [V3] Remove `cross-fetch` from `template-retail-react-app` (#1160)
  [v3] Add suffix to ssr build files (#1158)
  Consume new version of the commerce-sdk-isomorphic to enable Phased Launches support HTTP Basic Auth (#1153)

# Conflicts:
#	package-lock.json
#	packages/commerce-sdk-react/package-lock.json
#	packages/internal-lib-build/package-lock.json
#	packages/my-extended-retail-app/package-lock.json
#	packages/pwa-kit-create-app/package-lock.json
#	packages/pwa-kit-dev/package-lock.json
#	packages/pwa-kit-react-sdk/package-lock.json
#	packages/pwa-kit-runtime/package-lock.json
#	packages/template-retail-react-app/app/hooks/use-einstein.test.js
#	packages/template-retail-react-app/package-lock.json
#	packages/template-typescript-minimal/package-lock.json
#	packages/test-commerce-sdk-react/package-lock.json

bfeister committed:
lint:fix

vmarta committed:
Explore publishing certain packages only

vmarta committed:
Revert "fake release of retail-react-app v1"

This reverts commit f8dcaa0bbf4be7be7ef1d663170510c44ad71205.

vmarta committed:
Revert "fake sdk release v3"

This reverts commit 3aad2dabf33ed8e257354425e15e4b6cb217a53c.

bfeister committed:
fix failing tests in CI re: `TextDecoder is not defined`

bfeister committed:
take 2 attempting to fix TextDecoder global in jest

bfeister committed:
fix jest module mapper

bfeister committed:
fix jest moduleMapper config

bfeister committed:
REALLY fix jest moduleNameMapper config for new pathing + fix node `util` global TextDecoder is not defined in jest

bfeister committed:
add brand-logo

bfeister committed:
remove previously added `AbovePDP` Template Hook component

bfeister committed:
replay #1128 from bad merge

bfeister committed:
eslint:fix for `retail-react-app` base paths

bfeister committed:
de-duplicate deps in bundle

bfeister committed:
fix path defaulting, don't pass nullish value to path.resolve

bfeister committed:
improve pathing for overridesDir in pwa-kit-dev webapack config + add missing worker/main.js

bfeister committed:
Merge branch 'v3' into feature/template-extensibility-algo-refactor

* v3:
  [v3] Add Convenience Components (#1183)
  Update commerce-sdk-react README (#1176)
  Fix missing commerce-sdk-react logout call (#1180)

# Conflicts:
#	packages/template-retail-react-app/app/components/_app/index.jsx
#	packages/template-retail-react-app/app/pages/product-list/index.jsx

bfeister committed:
fix pathing

bfeister committed:
fix pathing issue that was mysteriously UNFIXED

bfeister committed:
fix `worker.js` path

vmarta committed:
Merge branch 'feature/template-extensibility-algo-refactor' into versioning-retail-react-app

vmarta committed:
Now also increment the pre-release version of the templates

vmarta committed:
Now allows for publishing one package only to npm

bfeister committed:
fix app/static pathing in non-extensible builds

vmarta committed:
Make sure Lerna uses the exact versions (without carrot)

vmarta committed:
Update package.json

vmarta committed:
What I expect to see from Ben's work

vmarta committed:
Some refactoring

vmarta committed:
Scale to multiple independent packages

vmarta committed:
Make sure version updates are synced throughout

vmarta committed:
Some refactoring

vmarta committed:
Remove commented lines

vmarta committed:
Merge branch 'feature/template-extensibility-algo-refactor' into versioning-retail-react-app

vmarta committed:
No need to do version-pinning ourselves

Because we now call `lerna version --exact`.

vmarta committed:
Add eslint and prettier

vmarta committed:
Move script to allow reuse by future templates

vcua-mobify committed:
[Extensibility] Remove Einstein proxy and send Einstein events directly from app (#1202)

* Remove Einstein proxy and send Einstein events directly from app

* Update ssrParameters

vmarta committed:
Merge branch 'feature/template-extensibility-algo-refactor' into versioning-retail-react-app

vcua-mobify committed:
Replace ExtendsCircularImportsPlugin with OverridesResolverPlugin (#1211)

* Replace ExtendsCircularImportsPlugin with OverridesResolverPlugin

* fix filepath problem

---------

Co-authored-by: Brian Feister <bfeister@salesforce.com>

vmarta committed:
Move `npm install` to the publishing step

vmarta committed:
Update console log

vmarta committed:
Prep for fixing a bug

vmarta committed:
Updating a package version checks deps in other packages

bfeister committed:
remove irrelevant bundlesize check to pass CI

vmarta committed:
Extract shared util functions

vmarta committed:
Consistency: add newline to EOF

vmarta committed:
Get the name of the current package

bfeister committed:
Fix `vendor.js` inclusion of `retail-react-app` files (#1198)

* fix duplicate lilbs in bundle

* fix app/static pathing for non-extensible builds

* oof, fix extensible app/static filepaths

* Resolve conflicting npm deps via `resolve.alias`

* add missing worker/main.js

* cleanup unused imports

* WIP, add list of original overridable deps

* prevent template extensibility vendor.js from treating baseline route files as deps in vendor.js

* fix irrelevant bundlesize declaration to pass CI

* drop test:max-file-size test from my-extended-retail-app

vmarta committed:
Merge branch 'feature/template-extensibility-algo-refactor' into versioning-retail-react-app

alexvuong committed:
replace relative with absolute path fetchTranslations (#1215)

Co-authored-by: Brian Feister <47546998+bfeister@users.noreply.github.com>

vmarta committed:
For now, make this package private

It's going to be deleted anyways in the near-future.

vmarta committed:
Iterate over the other packages only

vmarta committed:
Exclude these npm scripts from the smoke test

vmarta committed:
Handle scenario of an SDK package with independent version

bfeister committed:
Refactor for unit testing, fix pathing issue from old version of file parsing algoritihim

vmarta committed:
Add a 'sync' version of the script

vmarta committed:
Tweak conditions

vmarta committed:
Some refactoring

vmarta committed:
Some refactoring

vmarta committed:
Script to run a local npm registry

vmarta committed:
Add todos for bug fixes

vmarta committed:
Verify where the npm registry is

vmarta committed:
Merge branch 'feature/template-extensibility-algo-refactor' into versioning-retail-react-app

vmarta committed:
Ignore this linting error

vmarta committed:
Update regex

vmarta committed:
Add comments

vmarta committed:
Checks for clean working tree

vmarta committed:
Handle uncommitted changes

vmarta committed:
Make sure to update the package lock files

vmarta committed:
If error, make sure to still clean up

vmarta committed:
Install from the root

vmarta committed:
Some refactoring

vmarta committed:
No longer needing this install step

Now that we've run install in earlier steps.

vmarta committed:
For safety

vmarta committed:
If not on CI, make sure to publish to local npm

In order to avoid accidental publishing to remote npm registry.

vmarta committed:
Only exit if there's an error

bfeister committed:
Merge branch 'develop' into feature/template-extensibility

* develop: (37 commits)
  [Phased Launch] Call Session bridge after login (#1220)
  [v3] Add multi-site suffix to auth token keys (#1208)
  Footer: fix hydration error with the locale dropdown (#1210)
  remove unused peerDependency @chakra-ui/system (#1212)
  @W-12582733: Expose env vars endpoint for E2E smoke tests (#1207)
  Upgrade to React 18 (#1166)
  Remove v3 branch name related actions (#1206)
  add test to reach test coverage threshold
  remove commerce-api folder
  Feature: Extract einstein RefArch-specific values to constant (#1200)
  Bump version number to 2.7.1 and update changelogs (#1197)
  store usid in cookies (#1193)
  make sure static files are copied on dev environment (#1196)
  [WIP] PWA Kit 2.7.1 release (#1181)
  [V2] Internal lib build typescript dev dependency (#1194)
  [V2] Re-generate lock files and fix hook lib tests (#1186)
  [v3] Add Convenience Components (#1183)
  Update commerce-sdk-react README (#1176)
  Add additional properties to ShopperLogin test types (#1185)
  Fix missing commerce-sdk-react logout call (#1180)
  ...

# Conflicts:
#	package-lock.json
#	packages/commerce-sdk-react/CHANGELOG.md
#	packages/commerce-sdk-react/package-lock.json
#	packages/internal-lib-build/package-lock.json
#	packages/pwa-kit-create-app/package-lock.json
#	packages/pwa-kit-dev/package-lock.json
#	packages/pwa-kit-react-sdk/package-lock.json
#	packages/pwa-kit-runtime/package-lock.json
#	packages/template-mrt-reference-app/package-lock.json
#	packages/template-retail-react-app/package-lock.json
#	packages/template-typescript-minimal/package-lock.json
#	packages/test-commerce-sdk-react/package-lock.json

bfeister committed:
remove demo extensible app in light of soon-to-be-merged PR from @bendvc

bfeister committed:
lockfiles from reaact18 / chakra2

bfeister committed:
Feature/template extensibility (#1162)

* initial work on moving code to use resolver

* fix getoverridepath and break on ^ imports

* add todos and comments

* rewrite requestcontext.request for path with caret

* perform overrideshashmap lookup for relative import

* progress

* progress

* regenerate lock files

* fix dupe useToast

* progress on making `charles-solution` work

* initial pass at migration to `resolver.getHook('resolve').tapAsync` working, TODO next... cleanup

* restore icons file and add extensible text to home

* add some comments and construct hashmap in new plugin

* working glob sync lookup??

* resolve lookup when there is no extension

* add component slot for testing

* fix some lookup errors

* cleanup

* rewrite requests coming from underlying template

* cleanup

* rename plugin file

* remove unnecessary build files

* rename to overrides

* fix lookup errors

* fix some broken upstream changes, eslint fixes

* Bring back the deleted package.json

* Install new eslint plugin

* No more relative imports in retail-react-app

* Create .prettierrc.yaml

* restore other deleted package lockfiles

* fix some eslint issues

* remove `^` imports and begin migrating logic away from previous API. TODO: get webpack to resolve `retail-react-app` base import path from within `template-retail-react-app`

* short circuit plugin temporarily, get webpack `alias` working for template to resolve self-referencing `retail-react-app` imports

* fix unreachable condition

* wow... freaking works 🚀

* cleanup, get `template-retail-react-app` working again

* cleanup

* Create .eslintrc.js

* fix routes.jsx which no longer pulled relative template

* make routes.jsx more concise

* update lockfiles

* add jsconfig.json for IDEs to have magic `retail-react-app` be discoverable

* more cleanup

* change package.json key from `mobify` => `ccExtensibility`

* cleanup

* Final cleanup, remove console.logs, make `extendable` dynamic for internal file resolution in `template-retail-react-app` (and future projects), remove some non-required files,

* fix straggler non-dynamic filepath

* cleanup

* remove code that shouldn't be in v3+

* sync package.json deps in extended example project

* cleanup / pr feedback

* pr feedback, great catch @kevinxh

* fix problem with bad copy from underlying template file

* pr feedback / cleanup

* cleanup

Co-authored-by: Kevin He <kevin.he@salesforce.com>

* Update package-lock.json

* WIP

* drop unneeded complexity

* lockfiles

* moving toward an array of `alias`es

* Refactor template extensibility functionality to use `resolve.alias` + greatly reduce code footprint + make leading slash optional in `ccExtensibility.overridesDir`

* add brand-logo with no clash for server side viewBox

* add `request-processor.js` (required for worker.js to load) and fix bad extensibility path loading in `build-dev-server.js`

* lint fix

* fix failing pwa-kit-react-sdk tests

* cleanup

* lint:fix

* fix failing tests in CI re: `TextDecoder is not defined`

* take 2 attempting to fix TextDecoder global in jest

* fix jest module mapper

* fix jest moduleMapper config

* REALLY fix jest moduleNameMapper config for new pathing + fix node `util` global TextDecoder is not defined in jest

* add brand-logo

* remove previously added `AbovePDP` Template Hook component

* replay #1128 from bad merge

* eslint:fix for `retail-react-app` base paths

* de-duplicate deps in bundle

* fix path defaulting, don't pass nullish value to path.resolve

* improve pathing for overridesDir in pwa-kit-dev webapack config + add missing worker/main.js

* fix pathing

* fix pathing issue that was mysteriously UNFIXED

* fix `worker.js` path

* fix app/static pathing in non-extensible builds

* [Extensibility] Remove Einstein proxy and send Einstein events directly from app (#1202)

* Remove Einstein proxy and send Einstein events directly from app

* Update ssrParameters

* Replace ExtendsCircularImportsPlugin with OverridesResolverPlugin (#1211)

* Replace ExtendsCircularImportsPlugin with OverridesResolverPlugin

* fix filepath problem

---------

Co-authored-by: Brian Feister <bfeister@salesforce.com>

* remove irrelevant bundlesize check to pass CI

* Fix `vendor.js` inclusion of `retail-react-app` files (#1198)

* fix duplicate lilbs in bundle

* fix app/static pathing for non-extensible builds

* oof, fix extensible app/static filepaths

* Resolve conflicting npm deps via `resolve.alias`

* add missing worker/main.js

* cleanup unused imports

* WIP, add list of original overridable deps

* prevent template extensibility vendor.js from treating baseline route files as deps in vendor.js

* fix irrelevant bundlesize declaration to pass CI

* drop test:max-file-size test from my-extended-retail-app

* replace relative with absolute path fetchTranslations (#1215)

Co-authored-by: Brian Feister <47546998+bfeister@users.noreply.github.com>

* Refactor for unit testing, fix pathing issue from old version of file parsing algoritihim

* fix merge conflict resolution

* remove template extensible demo in prep for @bendvc's pr for generator work

* regen lockfiles for react 18 changes

* fix linting

---------

Co-authored-by: yunakim714 <yunakim@salesforce.com>
Co-authored-by: Vincent Marta <vmarta@salesforce.com>
Co-authored-by: Kevin He <kevin.he@salesforce.com>
Co-authored-by: vcua-mobify <vcua@salesforce.com>
Co-authored-by: vcua-mobify <47404250+vcua-mobify@users.noreply.github.com>
Co-authored-by: Alex Vuong <52219283+alexvuong@users.noreply.github.com>
Co-authored-by: Alex Vuong <alex.vuong@salesforce.com>

bfeister committed:
support windows file paths

bendvc committed:
Fix issue with failed tests because of react query dev tools

bfeister committed:
Merge branch 'feature/template-extensibility' into versioning-retail-react-app

* feature/template-extensibility: (38 commits)
  support windows file paths
  Feature/template extensibility (#1162)
  lockfiles from reaact18 / chakra2
  remove demo extensible app in light of soon-to-be-merged PR from @bendvc
  [Phased Launch] Call Session bridge after login (#1220)
  [v3] Add multi-site suffix to auth token keys (#1208)
  Footer: fix hydration error with the locale dropdown (#1210)
  remove unused peerDependency @chakra-ui/system (#1212)
  @W-12582733: Expose env vars endpoint for E2E smoke tests (#1207)
  Upgrade to React 18 (#1166)
  Remove v3 branch name related actions (#1206)
  add test to reach test coverage threshold
  remove commerce-api folder
  Feature: Extract einstein RefArch-specific values to constant (#1200)
  Bump version number to 2.7.1 and update changelogs (#1197)
  store usid in cookies (#1193)
  make sure static files are copied on dev environment (#1196)
  [WIP] PWA Kit 2.7.1 release (#1181)
  [V2] Internal lib build typescript dev dependency (#1194)
  [V2] Re-generate lock files and fix hook lib tests (#1186)
  ...

# Conflicts:
#	packages/my-extended-retail-app/config/default.js
#	packages/my-extended-retail-app/config/sites.js
#	packages/my-extended-retail-app/package-lock.json
#	packages/my-extended-retail-app/package.json
#	packages/my-extended-retail-app/pwa-kit-overrides/app/assets/svg/brand-logo.svg
#	packages/my-extended-retail-app/pwa-kit-overrides/app/components/_app-config/index.jsx
#	packages/my-extended-retail-app/pwa-kit-overrides/app/components/_app/index.jsx
#	packages/my-extended-retail-app/pwa-kit-overrides/app/pages/home/index.jsx
#	packages/my-extended-retail-app/pwa-kit-overrides/app/routes.jsx
#	packages/my-extended-retail-app/pwa-kit-overrides/app/ssr.js
#	packages/my-extended-retail-app/pwa-kit-overrides/app/static/ico/favicon.ico
#	packages/my-extended-retail-app/pwa-kit-overrides/app/static/img/global/app-icon-192.png
#	packages/my-extended-retail-app/pwa-kit-overrides/app/static/img/global/app-icon-512.png
#	packages/my-extended-retail-app/pwa-kit-overrides/app/static/img/global/apple-touch-icon.png
#	packages/pwa-kit-dev/bin/pwa-kit-dev.js
#	packages/pwa-kit-dev/package-lock.json
#	packages/pwa-kit-dev/package.json
#	packages/pwa-kit-dev/src/configs/webpack/config.js
#	packages/template-retail-react-app/app/components/confirmation-modal/index.test.js
#	packages/template-retail-react-app/app/components/footer/index.jsx
#	packages/template-retail-react-app/app/components/header/index.jsx
#	packages/template-retail-react-app/app/components/list-menu/index.test.js
#	packages/template-retail-react-app/app/components/product-scroller/index.test.js
#	packages/template-retail-react-app/app/components/product-view-modal/index.test.js
#	packages/template-retail-react-app/app/components/search/index.test.js
#	packages/template-retail-react-app/app/hoc/with-registration/index.test.js
#	packages/template-retail-react-app/app/hooks/use-add-to-cart-modal.js
#	packages/template-retail-react-app/app/hooks/use-auth-modal.test.js
#	packages/template-retail-react-app/app/hooks/use-currency.test.js
#	packages/template-retail-react-app/app/hooks/use-multi-site.test.js
#	packages/template-retail-react-app/app/hooks/use-navigation.test.js
#	packages/template-retail-react-app/app/pages/account/addresses.test.js
#	packages/template-retail-react-app/app/pages/account/index.jsx
#	packages/template-retail-react-app/app/pages/account/wishlist/index.test.js
#	packages/template-retail-react-app/app/pages/account/wishlist/partials/wishlist-primary-action.test.js
#	packages/template-retail-react-app/app/pages/cart/index.test.js
#	packages/template-retail-react-app/app/pages/cart/partials/cart-secondary-button-group.test.js
#	packages/template-retail-react-app/app/pages/checkout/index.test.js
#	packages/template-retail-react-app/app/pages/checkout/partials/contact-info.jsx
#	packages/template-retail-react-app/app/pages/checkout/partials/contact-info.test.js
#	packages/template-retail-react-app/app/pages/home/index.test.js
#	packages/template-retail-react-app/app/pages/product-list/partials/empty-results.jsx
#	packages/template-retail-react-app/app/pages/product-list/partials/page-header.jsx
#	packages/template-retail-react-app/app/pages/registration/index.test.jsx
#	packages/template-retail-react-app/app/utils/site-utils.js
#	packages/template-retail-react-app/package-lock.json
#	packages/template-retail-react-app/package.json

vmarta committed:
Remove package

vmarta committed:
Make sure versions are all 3.0.0-dev

vmarta committed:
Update package lock files

vmarta committed:
Follow up to recent merge conflicts

vmarta committed:
Revert these changes to the generator

vmarta committed:
Let the output of `npm version` pass through

vmarta committed:
Revert "Let the output of `npm version` pass through"

This reverts commit 33b3058ca4633fc0648559bc72a729ba262732d9.

vmarta committed:
Make sure to `npm install` after updating dependencies

vmarta committed:
Prepare to consolidate all of the versioning scripts

And move them to the root of monorepo.

vmarta committed:
Move the rest of scripts into the monorepo root

vmarta committed:
commerce-sdk-react now has its own version

vmarta committed:
Use correct package name

vmarta committed:
Add todos

vmarta committed:
Reconsider where to do `npm install`

vmarta committed:
Clean up

vmarta committed:
Remove todo

vmarta committed:
Merge branch 'develop' into versioning-retail-react-app

vmarta committed:
Remove unused dependency

vmarta committed:
Some code cleanup

vmarta committed:
Merge branch 'develop' into versioning-retail-react-app

vmarta committed:
Move the versioning scripts into their own folder

vmarta committed:
Merge branch 'develop' into versioning-retail-react-app

vmarta committed:
Namespace all public packages with '@salesforce'

vmarta committed:
Point to @salesforce/commerce-sdk-react

vmarta committed:
Point to @salesforce/pwa-kit-dev

vmarta committed:
Merge branch 'develop' into prep-v3-preview

vmarta committed:
Point to @salesforce/pwa-kit-react-sdk

vmarta committed:
Point to @salesforce/pwa-kit-runtime

vmarta committed:
Point to @salesforce/pwa-kit-create-app

vmarta committed:
Update package locks

vmarta committed:
Linting fix

vmarta committed:
commerce-sdk-react@1.0.0-dev

vmarta committed:
retail-react-app@1.0.0-dev

vmarta committed:
Update `bin` field

vmarta committed:
Update action.yml

bendvc committed:
Fix source for templates on npm

vmarta committed:
Comment out the check-dependencies.js for now

bendvc committed:
Update "extendable" value

bendvc committed:
Lint

bendvc committed:
Merge branch 'prep-v3-preview' of https://github.com/SalesforceCommerceCloud/pwa-kit into prep-v3-preview

bfeister committed:
replay @hajinsuha1 changes from #1219 where CI checks were stuck due to being an external contributor (#1237)

bendvc committed:
Fix jest module mapper

bendvc committed:
Fix retail test app using wrong source and id

vmarta committed:
Merge branch 'develop' into prep-v3-preview

vmarta committed:
SDK@3.0.0-preview.0

vmarta committed:
Handle salesforce namespace in package name

vmarta committed:
Merge branch 'prep-v3-preview' into sdk@3.0.0-preview.0

bendvc commented on packages/commerce-sdk-react/CHANGELOG.md:
```diff
@@ -1,3 +1,4 @@
+## v1.0.0-dev.0 (May 30, 2023)
 ## v1.0.0-dev (May 29, 2023)
 ## v3.0.0-dev (May 12, 2023)
```
Should be be getting rid of these 3.0.0 reference

bendvc commented:
@vmarta have you thought about what we are going to do with the change log for the retail react app? Since it's being re-versioned at 1.0.0.. does it make sense that we have all those old change updates under older versions? Not really a blocker tho.

vmarta committed:
Prepare for v3 preview release - @W-13481576@ (#1233)

* rewrite requestcontext.request for path with caret

* perform overrideshashmap lookup for relative import

* progress

* fix conflicts

* fix package lockfiles that were out of sync

* Revert "fix package lockfiles that were out of sync"

This reverts commit 7f3c3645ef963ba4d7a7c219e66ec93ac6fe2269.

* progress

* fix out of sync lockfiles

* merge conflict

* resolve merge conflict

* update spike project's engines / regen v3 lockfiles

* regenerate lock files

* add missing node engines

* add missing useToast (bad merge)

* updates from underlying template-retail-react-app

* drop unnecessary pages/product-detail override

* fix dupe useToast

* progress on making `charles-solution` work

* initial pass at migration to `resolver.getHook('resolve').tapAsync` working, TODO next... cleanup

* restore icons file and add extensible text to home

* add some comments and construct hashmap in new plugin

* working glob sync lookup??

* resolve lookup when there is no extension

* add component slot for testing

* fix some lookup errors

* migrate to new directory

* remove unnecessary directory nesting, remove unnecessary files for "minimum file project"

* add example of extending routes non-destructively, add example of quickly toggling some base constant values (e.g. categories shown on home page)

* Update example route

* add some demo examples of overrides

* cleanup

* commit package name change

* rewrite requests coming from underlying template

* cleanup

* rename plugin file

* remove unnecessary build files

* rename to overrides

* fix lookup errors

* fix some broken upstream changes, eslint fixes

* fix bad merge conflict

* Bring back the deleted package.json

* Install new eslint plugin

* No more relative imports in retail-react-app

* Create .prettierrc.yaml

* restore other deleted package lockfiles

* fix some eslint issues

* remove `^` imports and begin migrating logic away from previous API. TODO: get webpack to resolve `retail-react-app` base import path from within `template-retail-react-app`

* bring back deleted package lockfiles

* short circuit plugin temporarily, get webpack `alias` working for template to resolve self-referencing `retail-react-app` imports

* fix unreachable condition

* wow... freaking works 🚀

* cleanup, get `template-retail-react-app` working again

* cleanup

* Create .eslintrc.js

* fix routes.jsx which no longer pulled relative template

* make routes.jsx more concise

* update lockfiles

* add jsconfig.json for IDEs to have magic `retail-react-app` be discoverable

* more cleanup

* change package.json key from `mobify` => `ccExtensibility`

* cleanup

* Final cleanup, remove console.logs, make `extendable` dynamic for internal file resolution in `template-retail-react-app` (and future projects), remove some non-required files,

* fix straggler non-dynamic filepath

* cleanup

* remove code that shouldn't be in v3+

* bump-version script ignores the retail-react-app

* See if this `--pre-dist-tag` works as expected

* Clean up code

* Add todo

* sync package.json deps in extended example project

* cleanup / pr feedback

* Update npm scripts

* Disable `npm install` step

* List the packages' versions for easy review

* Fix bug with versioning the root package file

* A bit of refactoring

* Some clean up

* pr feedback, great catch @kevinxh

* fix problem with bad copy from underlying template file

* pr feedback / cleanup

* cleanup

Co-authored-by: Kevin He <kevin.he@salesforce.com>

* Update package-lock.json

* WIP

* drop unneeded complexity

* lockfiles

* moving toward an array of `alias`es

* Generator: try downloading retail-react-app@latest from npm

* Some refactoring

* Refactor template extensibility functionality to use `resolve.alias` + greatly reduce code footprint + make leading slash optional in `ccExtensibility.overridesDir`

* Tweak the versioning logic

* These are optional options

* Use a new npm tag for testing purpose

* add brand-logo with no clash for server side viewBox

* add `request-processor.js` (required for worker.js to load) and fix bad extensibility path loading in `build-dev-server.js`

* Add some comments

* Install shelljs

* retail-react-app: version updates scoped to this directory only

* Update versioning logic in the root

* retail-react-app: simplify the versioning logic

* fake sdk release v3

* fake release of retail-react-app v1

* lint fix

* fix failing pwa-kit-react-sdk tests

* cleanup

* regen package lockfiles

* lint:fix

* Explore publishing certain packages only

* Revert "fake release of retail-react-app v1"

This reverts commit f8dcaa0bbf4be7be7ef1d663170510c44ad71205.

* Revert "fake sdk release v3"

This reverts commit 3aad2dabf33ed8e257354425e15e4b6cb217a53c.

* fix failing tests in CI re: `TextDecoder is not defined`

* take 2 attempting to fix TextDecoder global in jest

* fix jest module mapper

* fix jest moduleMapper config

* REALLY fix jest moduleNameMapper config for new pathing + fix node `util` global TextDecoder is not defined in jest

* add brand-logo

* remove previously added `AbovePDP` Template Hook component

* replay #1128 from bad merge

* eslint:fix for `retail-react-app` base paths

* de-duplicate deps in bundle

* fix path defaulting, don't pass nullish value to path.resolve

* improve pathing for overridesDir in pwa-kit-dev webapack config + add missing worker/main.js

* fix pathing

* fix pathing issue that was mysteriously UNFIXED

* fix `worker.js` path

* Now also increment the pre-release version of the templates

* Now allows for publishing one package only to npm

* fix app/static pathing in non-extensible builds

* Make sure Lerna uses the exact versions (without carrot)

* Update package.json

* What I expect to see from Ben's work

* Some refactoring

* Scale to multiple independent packages

* Make sure version updates are synced throughout

* Some refactoring

* Remove commented lines

* No need to do version-pinning ourselves

Because we now call `lerna version --exact`.

* Add eslint and prettier

* Move script to allow reuse by future templates

* [Extensibility] Remove Einstein proxy and send Einstein events directly from app (#1202)

* Remove Einstein proxy and send Einstein events directly from app

* Update ssrParameters

* Replace ExtendsCircularImportsPlugin with OverridesResolverPlugin (#1211)

* Replace ExtendsCircularImportsPlugin with OverridesResolverPlugin

* fix filepath problem

---------

Co-authored-by: Brian Feister <bfeister@salesforce.com>

* Move `npm install` to the publishing step

* Update console log

* Prep for fixing a bug

* Updating a package version checks deps in other packages

* remove irrelevant bundlesize check to pass CI

* Extract shared util functions

* Consistency: add newline to EOF

* Get the name of the current package

* Fix `vendor.js` inclusion of `retail-react-app` files (#1198)

* fix duplicate lilbs in bundle

* fix app/static pathing for non-extensible builds

* oof, fix extensible app/static filepaths

* Resolve conflicting npm deps via `resolve.alias`

* add missing worker/main.js

* cleanup unused imports

* WIP, add list of original overridable deps

* prevent template extensibility vendor.js from treating baseline route files as deps in vendor.js

* fix irrelevant bundlesize declaration to pass CI

* drop test:max-file-size test from my-extended-retail-app

* replace relative with absolute path fetchTranslations (#1215)

Co-authored-by: Brian Feister <47546998+bfeister@users.noreply.github.com>

* For now, make this package private

It's going to be deleted anyways in the near-future.

* Iterate over the other packages only

* Exclude these npm scripts from the smoke test

* Handle scenario of an SDK package with independent version

* Refactor for unit testing, fix pathing issue from old version of file parsing algoritihim

* Add a 'sync' version of the script

* Tweak conditions

* Some refactoring

* Some refactoring

* Script to run a local npm registry

* Add todos for bug fixes

* Verify where the npm registry is

* Ignore this linting error

* Update regex

* Add comments

* Checks for clean working tree

* Handle uncommitted changes

* Make sure to update the package lock files

* If error, make sure to still clean up

* Install from the root

* Some refactoring

* No longer needing this install step

Now that we've run install in earlier steps.

* For safety

* If not on CI, make sure to publish to local npm

In order to avoid accidental publishing to remote npm registry.

* Only exit if there's an error

* remove demo extensible app in light of soon-to-be-merged PR from @bendvc

* lockfiles from reaact18 / chakra2

* Feature/template extensibility (#1162)

* initial work on moving code to use resolver

* fix getoverridepath and break on ^ imports

* add todos and comments

* rewrite requestcontext.request for path with caret

* perform overrideshashmap lookup for relative import

* progress

* progress

* regenerate lock files

* fix dupe useToast

* progress on making `charles-solution` work

* initial pass at migration to `resolver.getHook('resolve').tapAsync` working, TODO next... cleanup

* restore icons file and add extensible text to home

* add some comments and construct hashmap in new plugin

* working glob sync lookup??

* resolve lookup when there is no extension

* add component slot for testing

* fix some lookup errors

* cleanup

* rewrite requests coming from underlying template

* cleanup

* rename plugin file

* remove unnecessary build files

* rename to overrides

* fix lookup errors

* fix some broken upstream changes, eslint fixes

* Bring back the deleted package.json

* Install new eslint plugin

* No more relative imports in retail-react-app

* Create .prettierrc.yaml

* restore other deleted package lockfiles

* fix some eslint issues

* remove `^` imports and begin migrating logic away from previous API. TODO: get webpack to resolve `retail-react-app` base import path from within `template-retail-react-app`

* short circuit plugin temporarily, get webpack `alias` working for template to resolve self-referencing `retail-react-app` imports

* fix unreachable condition

* wow... freaking works 🚀

* cleanup, get `template-retail-react-app` working again

* cleanup

* Create .eslintrc.js

* fix routes.jsx which no longer pulled relative template

* make routes.jsx more concise

* update lockfiles

* add jsconfig.json for IDEs to have magic `retail-react-app` be discoverable

* more cleanup

* change package.json key from `mobify` => `ccExtensibility`

* cleanup

* Final cleanup, remove console.logs, make `extendable` dynamic for internal file resolution in `template-retail-react-app` (and future projects), remove some non-required files,

* fix straggler non-dynamic filepath

* cleanup

* remove code that shouldn't be in v3+

* sync package.json deps in extended example project

* cleanup / pr feedback

* pr feedback, great catch @kevinxh

* fix problem with bad copy from underlying template file

* pr feedback / cleanup

* cleanup

Co-authored-by: Kevin He <kevin.he@salesforce.com>

* Update package-lock.json

* WIP

* drop unneeded complexity

* lockfiles

* moving toward an array of `alias`es

* Refactor template extensibility functionality to use `resolve.alias` + greatly reduce code footprint + make leading slash optional in `ccExtensibility.overridesDir`

* add brand-logo with no clash for server side viewBox

* add `request-processor.js` (required for worker.js to load) and fix bad extensibility path loading in `build-dev-server.js`

* lint fix

* fix failing pwa-kit-react-sdk tests

* cleanup

* lint:fix

* fix failing tests in CI re: `TextDecoder is not defined`

* take 2 attempting to fix TextDecoder global in jest

* fix jest module mapper

* fix jest moduleMapper config

* REALLY fix jest moduleNameMapper config for new pathing + fix node `util` global TextDecoder is not defined in jest

* add brand-logo

* remove previously added `AbovePDP` Template Hook component

* replay #1128 from bad merge

* eslint:fix for `retail-react-app` base paths

* de-duplicate deps in bundle

* fix path defaulting, don't pass nullish value to path.resolve

* improve pathing for overridesDir in pwa-kit-dev webapack config + add missing worker/main.js

* fix pathing

* fix pathing issue that was mysteriously UNFIXED

* fix `worker.js` path

* fix app/static pathing in non-extensible builds

* [Extensibility] Remove Einstein proxy and send Einstein events directly from app (#1202)

* Remove Einstein proxy and send Einstein events directly from app

* Update ssrParameters

* Replace ExtendsCircularImportsPlugin with OverridesResolverPlugin (#1211)

* Replace ExtendsCircularImportsPlugin with OverridesResolverPlugin

* fix filepath problem

---------

Co-authored-by: Brian Feister <bfeister@salesforce.com>

* remove irrelevant bundlesize check to pass CI

* Fix `vendor.js` inclusion of `retail-react-app` files (#1198)

* fix duplicate lilbs in bundle

* fix app/static pathing for non-extensible builds

* oof, fix extensible app/static filepaths

* Resolve conflicting npm deps via `resolve.alias`

* add missing worker/main.js

* cleanup unused imports

* WIP, add list of original overridable deps

* prevent template extensibility vendor.js from treating baseline route files as deps in vendor.js

* fix irrelevant bundlesize declaration to pass CI

* drop test:max-file-size test from my-extended-retail-app

* replace relative with absolute path fetchTranslations (#1215)

Co-authored-by: Brian Feister <47546998+bfeister@users.noreply.github.com>

* Refactor for unit testing, fix pathing issue from old version of file parsing algoritihim

* fix merge conflict resolution

* remove template extensible demo in prep for @bendvc's pr for generator work

* regen lockfiles for react 18 changes

* fix linting

---------

Co-authored-by: yunakim714 <yunakim@salesforce.com>
Co-authored-by: Vincent Marta <vmarta@salesforce.com>
Co-authored-by: Kevin He <kevin.he@salesforce.com>
Co-authored-by: vcua-mobify <vcua@salesforce.com>
Co-authored-by: vcua-mobify <47404250+vcua-mobify@users.noreply.github.com>
Co-authored-by: Alex Vuong <52219283+alexvuong@users.noreply.github.com>
Co-authored-by: Alex Vuong <alex.vuong@salesforce.com>

* support windows file paths

* Remove package

* Make sure versions are all 3.0.0-dev

* Update package lock files

* Follow up to recent merge conflicts

* Revert these changes to the generator

* Fix issue with failed tests because of react query dev tools

* Let the output of `npm version` pass through

* Revert "Let the output of `npm version` pass through"

This reverts commit 33b3058ca4633fc0648559bc72a729ba262732d9.

* Make sure to `npm install` after updating dependencies

* Prepare to consolidate all of the versioning scripts

And move them to the root of monorepo.

* Move the rest of scripts into the monorepo root

* commerce-sdk-react now has its own version

* Use correct package name

* Add todos

* Reconsider where to do `npm install`

* Clean up

* Remove todo

* Remove unused dependency

* Some code cleanup

* Move the versioning scripts into their own folder

* Namespace all public packages with '@salesforce'

* Point to @salesforce/commerce-sdk-react

* Point to @salesforce/pwa-kit-dev

* Point to @salesforce/pwa-kit-react-sdk

* Point to @salesforce/pwa-kit-runtime

* Point to @salesforce/pwa-kit-create-app

* Update package locks

* Linting fix

* commerce-sdk-react@1.0.0-dev

* retail-react-app@1.0.0-dev

* Update `bin` field

* Update action.yml

* Fix source for templates on npm

* Comment out the check-dependencies.js for now

* Update "extendable" value

* Lint

* Fix jest module mapper

* Fix retail test app using wrong source and id

* Handle salesforce namespace in package name

* Add '@salesforce' to the prefix

---------

Co-authored-by: yunakim714 <yunakim@salesforce.com>
Co-authored-by: Brian Feister <bfeister@salesforce.com>
Co-authored-by: Brian Feister <47546998+bfeister@users.noreply.github.com>
Co-authored-by: Kevin He <kevin.he@salesforce.com>
Co-authored-by: vcua-mobify <vcua@salesforce.com>
Co-authored-by: vcua-mobify <47404250+vcua-mobify@users.noreply.github.com>
Co-authored-by: Alex Vuong <52219283+alexvuong@users.noreply.github.com>
Co-authored-by: Alex Vuong <alex.vuong@salesforce.com>
Co-authored-by: Ben Chypak <bchypak@salesforce.com>

vmarta committed:
Merge branch 'develop' into sdk@3.0.0-preview.0

vmarta committed:
Update package-lock.json
